packages/react-scripts/bin/react-scripts.js

```diff
@@ -13,6 +13,20 @@ case 'test':
     [require.resolve('../scripts/' + script)].concat(args),
     {stdio: 'inherit'}
   );
+  if (result.signal) {
+    if (result.signal == 'SIGKILL') {
+      console.log(
+        'This probably means the system ran out of memory or someone called ' +
+        '`kill -9` on the process.'
+      );
+    } else if (result.signal == 'SIGTERM') {
+      console.log(
+        'Someone might have called `kill` or `killall`, or the system could ' +
+        'be shutting down.'
+      );
+    }
```

Comment:
Let's prepend both messages with `The build failed because the process exited too early. `
Otherwise the context is not clear.